### ISSUE TITLE
Unable to re-render shiny interactive document 

### ISSUE THREAD

**Issue by blenback** [quarto-dev/quarto-cli#4105]
### Bug description

Hello, 
I am starting a simple shiny interactive document with a simple drop-down menu selection:

---
title: "LULCC_CH_model"
format: html
server: shiny
execute:
  echo: false
---

```{r setup, include=FALSE}
# Install packages
packs <- c("data.table", "shiny")
new.packs <- packs[!(packs %in% installed.packages()[, "Package"])]
if (length(new.packs)) install.packages(new.packs)

# Load required packages
invisible(lapply(packs, require, character.only = TRUE))

# Source custom functions
invisible(sapply(list.files("Scripts/Functions", pattern = ".R", full.names = TRUE, recursive = TRUE), source))
```

```{r}
selectInput("OS", "Operating system:",
                c("Windows", "GNU/Linux"))
textOutput("result")
```

```{r}
#| context: server
   output$result <- renderText({
     paste("You chose:", input$OS)
   })
```

```{ojs}
viewof flavor = Inputs.checkbox(["Salty", "Spicy", "Sour", "Umami"], {label: "Flavor"})
```

The 1st time I hit 'run document' the file renders with no problem when using either the viewer pane or window preview settings. However if I then make changes, save and hit 'run document' again. The preview disappears and is not replaced, the render tab displays the processing wheel to indicate something is happening but it just hangs and never updates. I get no error messages to the console and the last message printed to the render tab is: "Listening on http://127.0.0.1:7876" 

I ran quarto check in the terminal and everything came back ok also I do not have such problems re-rendering non-interactive .qmd documents. Nor do I have a problem when re-running a .qmd with just  interactive {ojs} chunks the problem seems to be limited to the shiny features. 

I have updated the shiny package and am runnnig the latest version of Quarto, Rstudio (2022.12.0 Build 353) and R (4.2.2 (2022-10-31 ucrt) on Windows 10.

Any help would be appreciated. 

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [ ] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by blenback** [comment]
Update: I just tried the same file on a colleagues computer and it gave the same behaviour i.e. re-rendering hanging and preview disappearing.

**Comment by cscheid** [comment]
Thanks for the report. Can you provide a self-contained example that we can run ourselves? Your document includes references to folders we don't see.

**Comment by blenback** [comment]
Apologies same premise but self contained: 

`---
title: "LULCC_CH_model"
format: html
server: shiny
execute:
  echo: false
---

````{r setup, include=FALSE}
# Install packages
library(shiny)
````

````{r}
selectInput("OS", "Operating system:",
                c("Windows", "GNU/Linux"))
textOutput("result")
````

````{r}
#| context: server
   output$result <- renderText({
     paste("You chose:", input$OS)
   })
````

````{ojs}
viewof flavor = Inputs.checkbox(["Salty", "Spicy", "Sour", "Umami"], {label: "Flavor"})
`````

**Comment by cscheid** [comment]
I do seem to be seeing something similar on OS X as well.

I found that you can hit the "stop" button on the right side of the pane before hitting "run document" again:

<img width="1532" alt="image" src="https://user-images.githubusercontent.com/285675/214909344-bc2f689f-a926-495b-9dff-07c09b89d42b.png">

I don't know if this is the intended behavior or if RStudio IDE should be automatically running "stop" before "Run Document" in those cases. I'll ask the RStudio IDE folks.

**Comment by blenback** [comment]
Thanks, just a note this approach with the "stop" button doesn't work for me on Windows.

**Comment by cderv** [comment]
Possible related issue that I am putting here for reference: https://github.com/rstudio/rstudio/issues/12546
Context: On windows, I can't even stop the shiny document.

**Comment by cderv** [comment]
This should be fixed in latest RStudio IDE daily version. @blenback could you try with the daily version ? 
You can find one at https://dailies.rstudio.com/ and install a portable version just for testing.

**Comment by cderv** [comment]
This is supposed to be fixed, but it seems there is another issue 

* https://github.com/rstudio/rstudio/issues/14039

Let's follow upstream

**Comment by gcefalu** [comment]
---
title: "LinearReg"
format: revealjs
editor: visual
profile: shiny
smaller: true
scrollable: true
---

{r}
#| label: libraries
#| echo: true
suppressWarnings(suppressMessages(library(shinylive)))
suppressWarnings(suppressMessages(library(shiny)))
suppressMessages(suppressWarnings(library(rmarkdown)))
suppressMessages(suppressWarnings(library(dplyr)))
suppressMessages(suppressWarnings(library(bslib)))

{r}
#| label: data
#| echo: true
#| message: true
#| warning: true

datasets <- list(Formaldehyde = Formaldehyde, Indometh = Indometh, 
                 LifeCycleSavings = LifeCycleSavings, OrchardSprays = OrchardSprays,                  Theoph = Theoph, chickwts = chickwts,faithful = faithful,                           randu = randu,ChickWeight = ChickWeight, Loblolly = Loblolly,                       anscombe = anscombe, beaver1 = beaver1, nottem = nottem,
                 stackloss = stackloss, swiss = swiss, DNase = DNase,
                 Nile = Nile, mtcars = mtcars, trees = trees)

{r}
#| label: selectinput
#| echo: true
#| message: true
#| warning: true

  
  titlePanel(tags$h2("DATA SETS", style =    "color: blue;"))
  radioButtons("inDs", "Select dataset:", choices = names(datasets))

{r}
#|context: server
ds <- reactive( datasets[[input$inDs]] )

{r}
varSelectInput("inVar", "Select variable:", datasets[[1]])
varSelectInput("inVar1", "Select a second variable:", datasets[[1]])
{r}
#|context: server
observe(updateVarSelectInput(session, "inVar", data = ds()))
observe(updateVarSelectInput(session, "inVar1", data = ds()))

I am writing a Quarto presentation with revealjs format and shiny for interactivity. The problem is that the panels diaplay, but whe a data set is selected using the redio buttons, the dropdown menu does not update the list of variables of the new dataset.  Could you please help with this?

**Comment by mcanouil** [comment]
@gcefalu 
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

@gcefalu it seems you are asking for help on an unrelated issue.
Please open a [Q&A Discussion](https://github.com/quarto-dev/quarto-web/discussions/categories/q-a) on the discussion board.
Be sure to read the guidelines.

**Comment by gcefalu** [comment]
I will. Thank you.Sent from my iPhoneOn Mar 11, 2026, at 2:28 PM, Mickaël Canouil ***@***.***> wrote:﻿mcanouil left a comment (quarto-dev/quarto-cli#4105)
@gcefalu
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, i.e., usually four ````.
See https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us.
